### DISCUSSION TITLE
Integrating zaje into `ble.sh`

### DISCUSSION THREAD

**Discussion by jessp01** [akinomyoga/ble.sh#350] [Ideas]
Hello maintainers,

First of all, excellent project:)

`zaje`  is a syntax highlighter that aims to cover all your shell colouring needs. It can act as an ad-hoc replacement for cat and, with a spot of one-line shell functions tail and other friends. I thought you might be interested in integrating it into `ble.sh`.

You can see it in action here:
[![asciicast](https://asciinema.org/a/597732.svg)](https://asciinema.org/a/597732)

it's also capable of of extracting text from an image:
[![asciicast](https://asciinema.org/a/599719.svg)](https://asciinema.org/a/599719)

I'm happy to submit a pull request is this is of interest.

Cheers,

**Comment by akinomyoga** [comment]
Are there any problems with enabling `zaje` in a ble.sh session? The reason that `fzf` and `zoxide` need integrations is that they cause problems when enabled in a ble.sh session. Presumably, a Bash configuration is supposed to work the same as in normal Bash sessions, so you can just run `. /etc/profile.d/zaje` in a ble.sh session the same as in a plain Bash.

**Reply by jessp01** [reply]
Hi @akinomyoga ,

There's no problem whatsoever:) I've been working with both for a while now. What I meant is that perhaps you'd be interested in installing `zaje` as part of the installation of `ble.sh`.

**Reply by akinomyoga** [reply]
Thanks for your reply and the suggestion. I'm afraid to say that I currently wouldn't be interested in bundling it as a part of the ble.sh installation. Ble is not a package manager, but one independent Bash configuration/plugin. Ble aims to provide fundamental parts of the shell interactivity that are hard to achieve without modifying the line editor or the shell itself. The features `zaje` aims at seem to be the ones that can actually be implemented without modifying the core part of ble.sh.

**Reply by jessp01** [reply]
Indeed, no need to change `ble.sh` at all. Like I said, I've been using both in tandem for some time now.
I just thought it would be handy for users to have out of the box as I feel it compliments `ble.sh`. 
Of course, it's your call and I respect your decision.

Cheers,
